### PR TITLE
Include fatal reason for pod pending events

### PR DESCRIPTION
**Pull request by wolfier**
It is unclear when a failed event in put into the watcher queue in the scenario where a pending pod has a container with a fatal contatiner state reason.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 1]
```diff
@@ -230,7 +230,6 @@ def process_status(
                 self.kube_config.worker_pod_pending_fatal_container_state_reasons
                 and "status" in event["raw_object"]
             ):
-                self.log.info("Event: %s Pending, annotations: %s", pod_name, annotations_string)
```
Comment (dirrao) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py:233]: why this line needs to be moved down?

Comment (wolfier) [reply airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py:233]: I forgot to add an else statement to the for loop. When there are no fatal reasons, the code path should be to print out the pod pending log.

**Comment by romsharon98** [comment]
Can we enter the elif and not print any log,therefore pass the else and still not print any log?

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 2]
```diff
@@ -250,10 +249,17 @@ def process_status(
                                 and container_status_state["waiting"]["message"] == "pull QPS exceeded"
                             ):
                                 continue
+                            self.log.error("Event: %s has container %s with fatal reason %s",
+                                    pod_name,
+                                    container_status["name"],
+                                    container_status_state["waiting"]["reason"]
+                            )
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py]:
```suggestion
                            self.log.error(
                                "Event: %s has container %s with fatal reason %s",
                                pod_name,
                                container_status["name"],
                                container_status_state["waiting"]["reason"],
                            )
```

**Review by uranusjr — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py [hunk 3]
```diff
@@ -250,10 +249,18 @@ def process_status(
                                 and container_status_state["waiting"]["message"] == "pull QPS exceeded"
                             ):
                                 continue
+                            self.log.error(
+                                "Event: %s has container %s with fatal reason %s",
```
Comment (pankajkoti) [inline airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py:253]: Will it always be a fatal reason when it reaches this place in the code? How about the waiting reason `ContainerCreating`?

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**